### PR TITLE
Master integration

### PR DESCRIPTION
**Pull request by BaherAbdelmalek**

### CODE REVIEW

**Comment by BaherAbdelmalek** [comment]
@Nabila63Ahmed

File: .idea/compiler.xml [hunk 1]
```diff
@@ -10,7 +10,7 @@
       </profile>
     </annotationProcessing>
     <bytecodeTargetLevel>
-      <module name="jobs" target="1.5" />
```
Comment (mostafa-abdullah) [inline .idea/compiler.xml]: Please put the .idea folder in gitignore and untrack it
```
git rm --cached -r .idea/
```

File: src/main/java/com/linkedin/replica/jobs/commands/impl/UserApplyForJobCommand.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+public class UserApplyForJobCommand extends Command {
+
+    public UserApplyForJobCommand(HashMap<String, Object> args) {
+        super(args);
+    }
+
+    public Object execute() throws IOException, SQLException {
+        validateArgs(new String[]{"userId","jobId"});
+        // get notifications from db
```
Comment (mostafa-abdullah) [inline src/main/java/com/linkedin/replica/jobs/commands/impl/UserApplyForJobCommand.java]: what notifications?

File: src/main/java/com/linkedin/replica/jobs/commands/impl/ViewAppliedJobsCommand.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+public class ViewAppliedJobsCommand extends Command {
+
+    public ViewAppliedJobsCommand(HashMap<String, Object> args) {
+        super(args);
+    }
+
+    public Object execute() throws IOException, SQLException {
+        validateArgs(new String[]{"userId"});
+        JobsHandler jobsHandler = (JobsHandler) this.dbHandler;;
+            ArrayList<ReturnedJob> jobs = jobsHandler.getAppliedJobs((String)args.get("userId"));
```
Comment (mostafa-abdullah) [inline src/main/java/com/linkedin/replica/jobs/commands/impl/ViewAppliedJobsCommand.java]: Please adjust the indentation

File: src/main/java/com/linkedin/replica/jobs/database/DatabaseInstance.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+ * A singleton class carrying a database instance
+ */
+
+import java.io.IOException;
+import java.sql.SQLException;
+
+import com.linkedin.replica.jobs.database.handlers.JobsHandler;
+import com.linkedin.replica.jobs.database.handlers.impl.ArangoSQLJobsHandler;
+
+public class DatabaseInstance {
```
Comment (mostafa-abdullah) [inline src/main/java/com/linkedin/replica/jobs/database/DatabaseInstance.java]: Where do you use this class?

File: src/main/java/com/linkedin/replica/jobs/database/DatabaseInstance.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+    // TODO uncouple arango, read from some config file
+    private static JobsHandler db;
+
+    static {
+        try {
```
Comment (mostafa-abdullah) [inline src/main/java/com/linkedin/replica/jobs/database/DatabaseInstance.java]: You shouldn't catch any exceptions here

File: src/main/java/com/linkedin/replica/jobs/database/handlers/JobsHandler.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import java.util.List;
+
+public interface JobsHandler extends DatabaseHandler{
+
+    public ArrayList<ReturnedJob> getAppliedJobs(String userID) throws SQLException;
+    public  ArrayList<ReturnedJob> getSavedJobs(String userId) throws SQLException;
+    public void createJobAsaCompany(HashMap<String, Object > args) throws SQLException;
+    public void deleteJobAsaCompany(String userId,String jobId);
+    public Job getJob(String JobID);
+    public void user_save_job(String userId,String jobId) throws SQLException;
```
Comment (mostafa-abdullah) [inline src/main/java/com/linkedin/replica/jobs/database/handlers/JobsHandler.java]: please use camel case convention :`userSaveJob`

File: src/main/java/com/linkedin/replica/jobs/database/handlers/JobsHandler.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    public void editJob(HashMap<String, Object > args);
+    public boolean RespondToJobsAsCompany(String userId,String jobId,String applicantId, int status) throws SQLException;
```
Comment (mostafa-abdullah) [inline src/main/java/com/linkedin/replica/jobs/database/handlers/JobsHandler.java]: please use camel case convention: `respondTo...`

File: src/main/java/com/linkedin/replica/jobs/database/handlers/impl/ArangoSQLJobsHandler.java [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+        System.out.println(CompanyId);
+        cursor = dbInstance.query(query1, bindVars, null, String.class);
+        String user = cursor.next();
+        if(user.equals(userId))
+            return true;
+            return false;
+    }
+
+
+    public boolean ValidateCompany(String userId, String companyId){
```
Comment (mostafa-abdullah) [inline src/main/java/com/linkedin/replica/jobs/database/handlers/impl/ArangoSQLJobsHandler.java]: `validateCompany` .. Camel case

File: src/main/java/com/linkedin/replica/jobs/models/lightUser.java [hunk 9]
```diff
@@ -1,12 +1,18 @@
-package models;
+package com.linkedin.replica.jobs.models;
+
+import com.arangodb.entity.DocumentField;
 
 public class lightUser {
+    @DocumentField(DocumentField.Type.KEY)
```
Comment (mostafa-abdullah) [inline src/main/java/com/linkedin/replica/jobs/models/lightUser.java]: Where to we use light models? If they are no longer needed please delete them. Also class names should be capitalized.

File: src/test/java/Core/ArangoHandlerTest.java [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+        databaseSeed = new DatabaseSeed();
+        arangoDb =DatabaseConnection.getInstance().getArangoDriver().
+                db(config.getAppConfigProp("db.name"));
+        databaseSeed.insertJobs();
+
+    }
+
+
+
+//    @Test
```
Comment (mostafa-abdullah) [inline src/test/java/Core/ArangoHandlerTest.java]: Why are all the tests commented?

File: target/classes/config/controller.config [hunk 11]
```diff
@@ -0,0 +1,19 @@
+controller.request.body.setmaxthreadcount=controller.func.max_thread_count
```
Comment (mostafa-abdullah) [inline target/classes/config/controller.config]: Please put the target in gitignore and untrack it

**Review by mostafa-abdullah — CHANGES_REQUESTED** [review]

**Review by mostafa-abdullah — APPROVED** [review]

**Pull request merged**